doc/rtd/topics/code_review.rst

```diff
@@ -0,0 +1,170 @@
+network.
+
+Goals
+=====
+
+This process has the following goals:
+
+* Ensure that cloud-init community members are receiving code reviews
+  in a timely fashion
+* Minimize the time the cloud-init core developers have to spend on the
+  parts of the code review process which aren't actual code review
+
+Roles
+=====
+
+For ease of understanding, we will refer to two different roles
+throughout the process:
+
+Proposer
+   The developer proposing a pull request (hereafter known as a PR).
+   This could be a core developer or community member.  (This is likely
+   the role you will be taking in this process.)
+
+the Committers
+   The group of cloud-init core developers.
+
+   In this process doc, cloud-init core developers are treated as a
+   bloc of interchangeable people.  This is a simplification, as in
+   reality we will need to manage transfer of reviews, dismissal of
+   stale reviews, parallel reviews from multiple committers, etc.  If
+   we find that handling these on an ad-hoc basis is causing problems,
+   then we should revisit and expand this process to include them.
+
+   (If the **Proposer** is a core developer, then they are not
+   considered part of this group for the purposes of this process.)
+
+Opening Phase
+=============
+
+In this phase, the **Proposer** is responsible for all actions. They
+should get a PR into good enough shape that it is worth **the
+Committers** spending time reviewing it.  Specifically, they are
+responsible for getting to a point where the continuous integration
+(CI) testing in Travis is passing.  Once a PR is passing CI, it moves
+into the :ref:`Review phase`.
+
+These are the steps that comprise the opening phase:
+
+1. The **Proposer** opens PR
+
+2. An automated comment will be added to the PR, outlining the steps
+   expected of the **Proposer** before a **Committer** will review
+   their PR.  (N.B. This automation is not yet implemented!)
+
+3. CI runs automatically, and if
+
+   CI fails
+      The **Proposer** is expected to fix CI failures.  If the
+      **Proposer** doesn't understand the nature of the failures they
+      are seeing, they should comment in the pull request to request
+      assistance.  Alternatively, for more immediate assistance, they
+      can ask in ``#cloud-init`` on the Freenode IRC network.
+
+   CI passes
+      Move on to the :ref:`Review phase`.
+
+Review Phase
+============
+
+In this phase, the **Proposer** and the **Committers** will iterate
+together to, hopefully, get the PR merged into the cloud-init codebase.
+There are three potential outcomes: merged, rejected permanently, and
+temporarily closed.
+
+These are the steps that comprise the review phase:
+
+1. **The Committers** assign a **Committer** to the PR
+
+   The cloud-init core developers will work amongst themselves to
+   ensure that this happens in a timely fashion.  (Generally,
+   assignment will happen at the team's daily internal meeting.)  Note
```

Comment:
Having "internal meetings" for an open source project is a bad idea. The meeting should be public, we have IRC for this it should be used.

Reply:
Broadly speaking, I agree. The internal meetings referred to here are the Canonical team's daily standup meetings, which happen via video call, and are our only regular daily touchpoint. `cloud-init` is only one of the projects that are covered by our team, and the other projects are not all open source.

If we want to reliably do this daily then, practically, we're going to be doing it at our internal meeting.

Reply:
Well, but that creates another barrier to entry to anyone that might aspire to become a core team member/committer and that does not happen to work at Canonical. While we do not have such a person it really makes not difference. But at the very least the wording chosen should reflect that possibility. I'd suggest something along these lines:

"""
PR reviews for final approval are picked up by core team members or assigned to core team members in a daily meeting.
"""

Not being specific about the "location" of the meeting allows for the possibility that an external to Canonical core team member can exist and can than participate in that meeting.

Reply:
Good advice, thank you!